OpenRA.Mods.Common/Traits/SupportPowers/SelectDirectionalTarget.cs

```diff
@@ -95,65 +99,39 @@ IEnumerable<Order> IOrderGenerator.Order(World world, CPos cell, int2 worldPixel
 					ExtraData = IsOutsideDragZone ? (uint)currentArrow.Direction.Facing : uint.MaxValue
 				};
 
+				mouseAttachment.Reset();
 				world.CancelInputMode();
 			}
 		}
 
 		void IOrderGenerator.Tick(World world)
 		{
-			targetCursor.Tick();
-
 			// Cancel the OG if we can't use the power
 			if (!manager.Powers.ContainsKey(order))
 				world.CancelInputMode();
 		}
 
 		bool IsOutsideDragZone
 		{
-			get { return dragStarted && (dragLocation - targetLocation).Length > 20; }
+			get { return dragStarted && dragDirection.Length > MinDragThreshold; }
 		}
 
 		IEnumerable<IRenderable> IOrderGenerator.Render(WorldRenderer wr, World world) { yield break; }
 
-		IEnumerable<IRenderable> IOrderGenerator.RenderAboveShroud(WorldRenderer wr, World world)
-		{
-			if (!activated)
-				return Enumerable.Empty<IRenderable>();
-
-			var targetPalette = wr.Palette(targetPlaceholderCursorPalette);
-
-			var location = activated ? targetLocation : Viewport.LastMousePos;
-			var worldPx = wr.Viewport.ViewToWorldPx(location);
-			var worldPos = wr.ProjectedPosition(worldPx);
-			var renderables = new List<IRenderable>(targetCursor.Render(worldPos, WVec.Zero, -511, targetPalette, 1 / wr.Viewport.Zoom));
-
-			if (IsOutsideDragZone)
-			{
-				var directionPalette = wr.Palette(directionArrowPalette);
-				renderables.Add(new SpriteRenderable(currentArrow.Sprite, worldPos, WVec.Zero, -511, directionPalette, 1 / wr.Viewport.Zoom, true));
-			}
-
-			if (hideMouse)
-			{
-				hideMouse = false;
-				Game.RunAfterTick(() => Game.HideCursor = true);
-			}
-
-			return renderables;
-		}
+		IEnumerable<IRenderable> IOrderGenerator.RenderAboveShroud(WorldRenderer wr, World world) { yield break; }
 
 		string IOrderGenerator.GetCursor(World world, CPos cell, int2 worldPixel, MouseInput mi) { return cursor; }
 
 		void IOrderGenerator.Deactivate()
 		{
 			if (activated)
-				Game.HideCursor = false;
+				Game.Cursor.Unlock();
```

Comment:
Adding
```csharp
// HACK: Reset the cursor manually before unlocking to avoid a graphical glitch
Game.Cursor.SetCursor("default");
```

seems to fix the jumping issue.